packages/mrl/src/getTransferData/getMoonChainData.ts

```diff
@@ -20,30 +23,40 @@ export async function getMoonChainData({
   }
 
   const moonChain = getMoonChain(route.source.chain);
+  const isDestinationMoonChain = moonChain.isEqual(route.destination.chain);
+  let address = isDestinationMoonChain ? destinationAddress : sourceAddress;
```

Comment:
please maybe comment somehow all these complex logic.

Reply:
I refactored the code and included comment to improve it's comprehension